packages/core/src/lib/session.ts

```diff
@@ -42,6 +45,55 @@ export const assignInteractionResults = async (
   ctx.body = { redirectTo, ts };
 };
 
+export const checkProtectedAccess = async (
+  ctx: Context,
+  provider: Provider,
+  lifetime = 10 * 60
+) => {
+  const { result } = await provider.interactionDetails(ctx.req, ctx.res);
+
+  if (!result?.login?.accountId) {
+    throw new RequestError('auth.unauthorized');
+  }
+
+  if (!result.login.ts || result.login.ts < dayjs().unix() - lifetime) {
+    const user = await findUserById(result.login.accountId);
+
+    throw new RequestError('auth.require_re_authentication', {
+      username: conditional(
+        user.username && maskUserInfo({ type: 'username', value: user.username })
+      ),
+      phone: conditional(
+        user.primaryPhone && maskUserInfo({ type: 'phone', value: user.primaryPhone })
+      ),
+      email: conditional(
+        user.primaryEmail && maskUserInfo({ type: 'email', value: user.primaryEmail })
+      ),
+    });
+  }
+};
+
+export const refreshSignInTimestamp = async (ctx: Context, provider: Provider) => {
```

Comment:
Where is this used?

Reply:
This will be used after re-authenticated, I can bring it in the next PR